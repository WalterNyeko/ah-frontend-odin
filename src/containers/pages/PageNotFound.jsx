import React from 'react';
import DefaultLayout from 'containers/layout/DefaultLayout';

const PageNotFound = () => (
  <DefaultLayout>
    <div className="container">
      <h2 className="text-center">The page you are looking for cannot be found</h2>
    </div>
  </DefaultLayout>
);

export default PageNotFound;
