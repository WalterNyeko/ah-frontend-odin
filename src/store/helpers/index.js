import { pick, isObjectLike, get, zipObject } from 'lodash';
import { getAllArticles } from 'store/actions/articles';

export const mapStateToProps = props => state => {
  if (Array.isArray(props)) return pick(state, props);

  if (isObjectLike(props)) {
    const keys = Object.keys(props);

    const values = keys.map(key => get(state, props[key]));
    return zipObject(keys, values);
  }

  return {};
};

export const getArticles = ({ articles }) => ({ articles: articles.all });

export const fetAllArticles = dispatch => ({ fetchAllArticles: () => dispatch(getAllArticles()) });
export const sortByUpdatedAt = data => {
  const toDate = object => {
    const updatedAt = object.updated_at || object.updatedAt;
    return (updatedAt && new Date(updatedAt)) || 0;
  };
  // sort articles by updated_at
  return data.sort((a, b) => toDate(b) - toDate(a));
};
export default mapStateToProps;
