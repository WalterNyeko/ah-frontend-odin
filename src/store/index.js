import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import Http from 'utils/Http';
import { API_URL } from 'utils/constants';

import rootReducer from './reducers';

const http = new Http(API_URL);
const shouldLog = process.env.NODE_ENV === 'development';
const middleware = [thunk.withExtraArgument(http)];

/* istanbul ignore if */
if (shouldLog) {
  middleware.push(logger);
}

const store = createStore(rootReducer, applyMiddleware(...middleware));

export default store;
