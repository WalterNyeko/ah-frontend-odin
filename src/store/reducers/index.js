import { combineReducers } from 'redux';
import UIReducer from './UIReducer';
import articles from './article';

export default combineReducers({
  UIReducer,
  articles,
});
