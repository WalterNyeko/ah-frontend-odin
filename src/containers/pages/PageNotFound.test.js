import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from 'store';
import PageNotFound from './PageNotFound';

describe('PageNotFound', () => {
  test('It renders the same UI', () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <MemoryRouter>
              <PageNotFound />
            </MemoryRouter>
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
