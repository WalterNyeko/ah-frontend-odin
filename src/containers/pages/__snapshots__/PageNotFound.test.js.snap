// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound It renders the same UI 1`] = `
<div
  className=""
>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary mb-2"
  >
    <a
      aria-current="page"
      className="navbar-brand d-flex align-items-center active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Author Havens Logo"
        src="logo.png"
      />
      <span
        className="h3 font-weight-bold  pl-1"
      >
         Authors Haven 
      </span>
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/articles"
            onClick={[Function]}
          >
            Articles
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="nav-link text-light btn-link"
            data-id="login"
            onClick={[Function]}
          >
            login
          </button>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="nav-link text-light btn-link"
            data-id="signup"
            onClick={[Function]}
          >
            Register
          </button>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="main"
  >
    <div
      className="container"
    >
      <h2
        className="text-center"
      >
        The page you are looking for cannot be found
      </h2>
    </div>
  </div>
</div>
`;
